From 7fd0caf9e2ff951e3fe7b6582d6274b2ec8ce600 Mon Sep 17 00:00:00 2001
From: WangJing <wangjing@mprc.pku.edu.cn>
Date: Wed, 26 Oct 2011 17:04:18 +0800
Subject: [PATCH 035/641] UniCore64: Add mem_init and free_initmem function

Fixed errors:
init/built-in.o: In function `init_post':
main.c:(.text+0x2a): undefined reference to `free_initmem'
init/built-in.o: In function `start_kernel':
(.init.text+0x4ea): undefined reference to `mem_init'

Signed-off-by: WangJing <wangjing@mprc.pku.edu.cn>
---
 arch/unicore64/mm/Makefile |    2 +-
 arch/unicore64/mm/init.c   |   19 +++++++++++++++++++
 2 files changed, 20 insertions(+), 1 deletion(-)
 create mode 100644 arch/unicore64/mm/init.c

diff --git a/arch/unicore64/mm/Makefile b/arch/unicore64/mm/Makefile
index cb53aed..2b21894 100644
--- a/arch/unicore64/mm/Makefile
+++ b/arch/unicore64/mm/Makefile
@@ -1 +1 @@
-obj-y				:= mmu.o pgtable.o ioremap.o
+obj-y				:= mmu.o pgtable.o ioremap.o init.o
diff --git a/arch/unicore64/mm/init.c b/arch/unicore64/mm/init.c
new file mode 100644
index 0000000..ddb7a60
--- /dev/null
+++ b/arch/unicore64/mm/init.c
@@ -0,0 +1,19 @@
+#include <linux/kernel.h>
+
+/**
+ * mem_init() -
+ */
+void __init mem_init(void)
+{
+	/* FIXME */
+	BUG();
+}
+
+/**
+ * free_initmem() -
+ */
+void free_initmem(void)
+{
+	/* FIXME */
+	BUG();
+}
-- 
1.7.9.5

